src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -185,14 +197,86 @@ def __unblock_jobs(cls):
                 with cls.lock:
                     try:
                         job.pending()
-                        cls.blocked_jobs.remove(job)
                         cls.jobs_to_run.put(job)
+                        cls.__remove_blocked_job(job)
                     except:
                         cls.__logger.warning(f"{job.id} is not in the blocked list anymore.")
 
+    @classmethod
+    def __remove_blocked_job(cls, job: Job):
+        if job in cls.blocked_jobs:
+            cls.blocked_jobs.remove(job)
+
+    @classmethod
+    def __remove_jobs_to_run(cls, jobs: Union[Job, List[Job], Set[Job]]):
+        jobs = jobs if isinstance(jobs, List) or isinstance(jobs, Set) else [jobs]
+        new_jobs_to_run: Queue = Queue()
+        while not cls.jobs_to_run.empty():
+            current_job = cls.jobs_to_run.get()
+            if current_job not in jobs:
+                new_jobs_to_run.put(current_job)
+        cls.jobs_to_run = new_jobs_to_run
+
     @classmethod
     def _update_job_config(cls):
         if Config.job_config.is_standalone:  # type: ignore
             cls._dispatcher = _StandaloneJobDispatcher()
         else:
             cls._dispatcher = _DevelopmentJobDispatcher()
+
+    @classmethod
+    def _set_process_in_scheduler(cls, job_id, process):
+        cls._processes[job_id] = process
+
+    @classmethod
+    def _pop_process_in_scheduler(cls, job_id, default=None):
+        return cls._processes.pop(job_id, default)  # type: ignore
+
+    @classmethod
+    def _cancel_job(cls, job):
+        to_cancel_jobs = set()
+        to_cancel_jobs.update([job])
+        to_cancel_jobs.update(cls.__find_subsequent_jobs(job.submit_id, set(job.task.output.keys())))
+        cls.__remove_jobs_from_running(to_cancel_jobs)
+        for job in to_cancel_jobs:
+            cls.__cancel_process(job)
+            job.cancelled()
+
+            # cls.__unlock_edit_on_outputs(job)
+
+        if not cls.jobs_to_run.empty():
+            cls.__execute_jobs()
+        else:
+            cls.__unblock_jobs()
+            cls.__execute_jobs()
```

Comment:
Why do you need to re execute some jobs after the cancelation? 
either there is a job already running that will trigger the next execution, either there is no running job that means there is nothing to run. AM I missing something here ?

Reply:
@jrobinAV When canceling a job, meaning the process pool will have 1 process available. Ideally, it will be occupied with the next job in the queue is my idea. Canceling a job, if possible, will cancel a process, which means the process was never done and cannot initiate the execution of the next job. If there's no current running job to initiate the next job when done, meaning if there's still some jobs left in the queue, there's nothing that will call the next job to initiate. I'm not sure if my explanation is clear enough. Do let me know what you think